Add a builder example to show the basics

tOkeshu opened the pull request:
I was playing with the `Builder` widget but could not make it work.
I decided to write a basic example so someone could help me fix what is going on and eventually add it to the example repo.

If you run `cargo build` you should see the following error:
```rust
src/builder_basics.rs:15:5: 15:22 error: the type of this value must be known in this context
src/builder_basics.rs:15     window.show_all();
                             ^~~~~~~~~~~~~~~~~
```

The glade file is taken from a [Python GTK tutorial](https://python-gtk-3-tutorial.readthedocs.org/en/latest/builder.html).

If we make this work, I should be able to add 1 or 2 examples showing how to use the builder.
Tell me what you think. :smile:

gkoz commented on Cargo.toml:
```diff
@@ -7,9 +7,11 @@ authors = ["The Rust-gnome Project Developers"]
 cairo-rs = "^0"
 
+[dependencies.gtk]
+features = ["gtk_3_10", "cairo_1_12"]
```
We can't accept changes that prevent us from building the examples crate with older versions of GTK.
As this whole example seems to depend on 3.10 I suggest not touching the features in `Cargo.toml` and putting the whole logic into a submodule gated on `gtk_3_10` like this:
```Rust
extern crate glib;
extern crate gtk;

#[cfg(feature = "gtk_3_10")]
mod a {
    use gtk;
    // ...
    pub fn main() {
        // ...
    }
}

#[cfg(feature = "gtk_3_10")]
fn main() { a::main() }

#[cfg(not(feature = "gtk_3_10"))]
fn main() { }
```

gkoz replied:
Alternatively, we could add the [`add_from_file`](https://developer.gnome.org/gtk3/stable/GtkBuilder.html#gtk-builder-add-from-file) method and avoid depending on 3.10 altogether.

tOkeshu replied:
While I would prefer to have `add_from_file`, I think that would be a good thing to have an example using GTK 3.10, so next contributors can have something they can copy for other examples requiring 3.10.

GuillaumeGomez commented:
@tOkeshu: Hey, thanks for your PR ! However, we need examples wich can build on travis and appveyor in order to see if your work is building properly. So please follow @gkoz instructions to update your code. I'll merge your PR once this is done.

PS: Post a comment to notify me.

gkoz commented:
> error: the type of this value must be known in this context

The compiler doesn't know what we're building and needs out help like this:
```Rust
let window: gtk::Window = builder.get_object("window1").unwrap();
```

Currently there doesn't seem to be any runtime checking if the type is the correct one. This will probably be fixed in the upcoming object wrappers reform outlined in https://github.com/rust-gnome/gtk/pull/59

tOkeshu commented:
I updated the patch to follow the module "scoped" to GTK 3.10.
I also modified a bit the example and updated the README.

Tell me if that's good for you @GuillaumeGomez and @gkoz :)

GuillaumeGomez commented:
@tOkeshu: Very good job ! It's perfectly fine for me. Is it good for you too @gkoz ?

gkoz commented on src/builder_basics.rs:
```diff
@@ -0,0 +1,41 @@
+//!
+//! This sample demonstrates how to use the builder with a simple glade file
+
+extern crate glib;
```
We don't need this import, do we? Let's not confuse anyone who might cargo-cult this into their own projects :)

gkoz commented:
Aside of the last small nit, :+1: good job.

tOkeshu commented:
Fixed.

Thank you for your help @GuillaumeGomez and @gkoz.

GuillaumeGomez commented:
No problem ! Thank you for contributing ! Don't hesitate to help more if you have time ! =D
